### PR TITLE
Various plot fixes

### LINKED ISSUE #1367: `log` scale button has wrong starting state if `norm='log'` is given
```python
sc.neutron.instrument_view(sample, norm='log', pixel_size=0.01, bins=1)
```
Clicking `log` once has no effect. Clicking again disables log.

### LINKED ISSUE #1368: Instrument view has `position` a a selectable slider option?
image

Getting some exception when clicking the button / moving the slider.

### LINKED ISSUE #1372: Thickness slider in plot of 3d data missing if unrelated MD coord present
```python
import numpy as np
import scipp as sc
var = sc.Variable(dims=['x', 'y', 'z'], values=np.random.rand(9,9,9))
a = sc.DataArray(data=var, coords={'label':var})
sc.plot.plot(a)
```

### PR DESCRIPTION
**Pull request by nvaytet**
See linked issues for more details.

Fixes #1372 
Fixes #1368 
Fixes #1367

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
New or old issues:
```python
plot(sample, norm='log', vmin=1, vmax=1000)
```
Now switch `log` off and on again. In both normal and log scale the originally given `vmin` and `vmax` seem to be ignored?

File: python/src/scipp/plot/figure3d.py [hunk 1]
```diff
@@ -405,3 +404,7 @@ def update_log_axes_buttons(self, *args, **kwargs):
         """
         if self.toolbar is not None:
             self.toolbar.update_log_axes_buttons(*args, **kwargs)
+
+    def update_norm_button(self, *args, **kwargs):
+        if self.toolbar is not None:
+            self.toolbar.update_norm_button(*args, **kwargs)
```
Comment (SimonHeybrock) [inline python/src/scipp/plot/figure3d.py:410]: Same as in `figure.py`? Should there be a common base class?

File: python/src/scipp/plot/plot3d.py [hunk 2]
```diff
@@ -78,6 +72,15 @@ def __init__(self,
...
                                  positions=positions)
 
+        # Create control widgets (sliders and buttons)
+        self.widgets = PlotWidgets(axes=self.axes,
+                                   ndim=view_ndims,
+                                   name=self.name,
+                                   dim_to_shape=self.dim_to_shape,
+                                   masks=self.masks,
+                                   positions=positions,
+                                   multid_coord=self.model.get_multid_coord())
```
Comment (SimonHeybrock) [inline python/src/scipp/plot/plot3d.py:82]: Almost identical code 3 times, is another common base class possible? Or maybe most of these arguments should be stored in a class or dict, such that the duplicate code is less, something like `self.widgets = PlotWidget(self.plot_params)`?

Not suggesting that this must be addressed in the PR, just something to think about for the next round of cleanup.

**Review by SimonHeybrock — COMMENTED** [review]
Apart from the issue I mentioned above, all three issues are fixed. 👍

File: python/src/scipp/plot/figure3d.py (hunk 1 above)
Comment (nvaytet) [reply python/src/scipp/plot/figure3d.py:410]: I already have quite a few base classes, I'm not sure I want to make a new one just for one function. I'm starting to feel that the more inheritance I add the less flexible everything becomes.

File: python/src/scipp/plot/plot3d.py (hunk 2 above)
Comment (nvaytet) [reply python/src/scipp/plot/plot3d.py:82]: Yes, I think I prefer the dict option. I suggest delaying to a later cleanup as not relevant to this PR

File: python/src/scipp/plot/sciplot.py [hunk 3]
```diff
@@ -216,19 +223,15 @@ def _process_axes_dimensions(self,
...
-    def _validate_axes(self, array):
+    def validate(self):
         """
-        Validation checks on axes.
+        Validation checks before plotting.
         """
-        for dim, coord in array.coords.items():
-            if len(coord.dims) > 1:
-                self.multid_coord = dim
+        multid_coord = self.model.get_multid_coord()
```
Comment (OwenArnold) [inline python/src/scipp/plot/sciplot.py:230]: You also have a `self.multid_coord` variable, which I think you never use? you could avoid the temporary and just make the function call as part of the test.

File: python/src/scipp/plot/widgets.py [hunk 4]
```diff
@@ -89,6 +89,11 @@ def __init__(self,
                 button_style="",
                 layout={"width": "initial"})
 
+            # TODO: hide the profile button for 3D plots. Renable this once
+            # profile picking is supported on 3D plots
+            if ndim == 3:
```
Comment (OwenArnold) [inline python/src/scipp/plot/widgets.py:94]: I'm not sure this logic belongs in the widget, but maybe you were going to address all this as some later fix?

**Comment by nvaytet** [comment]
> New or old issue:
> 
> ```python
> plot(sample, norm='log', vmin=1, vmax=1000)
> ```
> 
> Now switch `log` off and on again. In both normal and log scale the originally given `vmin` and `vmax` seem to be ignored?

I think this is it not from this PR. I need to feed the global `vmin/vmax` to the `rescale_to_data()` that is performed when we change from linear to log.
I am also not sure what happens if you want to set the y limits on a 1d plot.
Finally, i think i need to improve the exposure of matplotlib objects (axes and figure) inside the `Plot` object. As the https://scipp.github.io/visualization/customizing-figures.html notebook illustrates, `out['tof.counts'].view.figure.ax.set_title('This is a new title!')` is not a great API to work with.
I'll open a new issue with all these points.

File: python/src/scipp/plot/widgets.py (hunk 4 above)
Comment (nvaytet) [reply python/src/scipp/plot/widgets.py:94]: Yes, this is only meant to be temporary.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**